Sources/Parley/Data/Remotes/ParleyNetworkSession.swift

```diff
@@ -35,3 +35,34 @@ public protocol ParleyNetworkSession: Sendable {
         completion: @Sendable @escaping (_ result: Result<ParleyHTTPDataResponse, ParleyHTTPErrorResponse>) -> Void
     )
 }
+
+extension ParleyNetworkSession {
+    
+    public func request(
+        _ url: URL,
+        data: Data?,
+        method: ParleyHTTPRequestMethod,
+        headers: [String: String]
+    ) async -> Result<ParleyHTTPDataResponse, ParleyHTTPErrorResponse> {
```

Comment:
Would it not be more logical in the swift 6 context to make the signature like this? 
```suggestion
    ) async throws(ParleyHTTPErrorResponse) -> ParleyHTTPDataResponse {
```